azavea/district-builder-dtl-pa
# Add script to scrape HTML files and add data to submissions CSV

## Overview

C70 wants to include some metrics in the CSV we send them along with the submission reports. To do this, this solution takes a submissions CSV, scrapes the HTML submission report files, and builds a new CSV with additional columns for `competitiveness`, `split_counties`, `majority_minority`, `compactness`, and `population_equivalence`.

### Checklist

~- [ ] Files changed in the PR have been `yapf`-ed for style violations~

### Notes

This will likely run only once, on 2018-12-15 by @shreshthkhilani.


## Testing Instructions

 * Make sure you have a few submissions locally. If not, you can do these steps on staging. 
 * On the VM, enter the Django container `sudo docker exec -it <container_id> bash`. 
 * Generate submission reports `for i in {1..240}; do ./manage.py submission_report $i; done`
 * Enter the dbshell with `./manage.py dbshell` and run `\copy (SELECT * FROM redistricting_plansubmission) TO '/tmp/submissions.csv' WITH CSV HEADER`.
 * You should now have a `submissions.csv` and some `*.html`s
 * Outside the container, copy the csv and html files out with `sudo docker cp <container_id> <file_path> .`
 * Outside the VM, scp the files down with `scp varant@127.0.0:/home/vagrant/$i.html .`.
 * Now, build the go file `go build main.go` and run it `./main <path to csv> <path to dir with htmls>`
 * Check a few html files against the `output.csv` file to ensure that the values are correct.

Closes [#162639036](https://www.pivotaltracker.com/story/show/162639036)
